### PR TITLE
[HOPS-1514] Cleanup bad coding in metadata-dal-impl-ndb

### PR DESCRIPTION
**Pull request by berthoug**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added and passed (for bug fixes / features)
- [x] HOPS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit
- [x] The commit message has the following format: [HOPS-XXX] message

* **Post a link to the associated JIRA issue**
https://hopshadoop.atlassian.net/browse/HOPS-1514

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/BlockChecksumClusterj.java [hunk 1]
```diff
@@ -155,17 +155,23 @@ public BlockChecksum find(long inodeId, int blockIndex)
...
-      PreparedStatement s = conn.prepareStatement(query);
+      s = conn.prepareStatement(query);
       s.executeUpdate();
     } catch (SQLException ex) {
       throw HopsSQLExceptionHelper.wrap(ex);
     } finally {
+      if (s != null) {
+        try {
+          s.close();
+        } catch (SQLException ex) {
```
Comment (maismail) [inline src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/BlockChecksumClusterj.java]: shouldn't we throw the SQLException? or at least log it

File: src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/EncodingStatusClusterj.java [hunk 2]
```diff
@@ -574,6 +575,18 @@ private EncodingStatus createHopEncoding(EncodingStatusDto dto) {
     } catch (SQLException ex) {
       throw HopsSQLExceptionHelper.wrap(ex);
     } finally {
+      if (s != null) {
+        try {
+          s.close();
+        } catch (SQLException ex) {
```
Comment (maismail) [inline src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/EncodingStatusClusterj.java]: same exception issue

File: src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/QuotaUpdateClusterj.java [hunk 3]
```diff
@@ -170,6 +173,18 @@ public void prepare(Collection<QuotaUpdate> added,
...
+      if (s != null) {
+        try {
+          s.close();
+        } catch (SQLException ex) {
+        }
+      }
+      if (result != null) {
+        try {
+          result.close();
+        } catch (SQLException ex) {
```
Comment (maismail) [inline src/main/java/io/hops/metadata/ndb/dalimpl/hdfs/QuotaUpdateClusterj.java]: same exception issue

File: src/main/java/io/hops/metadata/ndb/dalimpl/yarn/RMLoadClusterJ.java [hunk 4]
```diff
@@ -88,16 +88,16 @@ private RMLoadDTO createPersistable(Load entry, HopsSession session)
...
       Load hop = createHopLoad(dto);
       map.put(hop.getRmHostName(), hop);
     }
     return map;
   }
 
   private Load createHopLoad(RMLoadDTO loadDTO) {
-    if (loadDTO == null) {
-      return null;
-    }
```
Comment (maismail) [inline src/main/java/io/hops/metadata/ndb/dalimpl/yarn/RMLoadClusterJ.java]: With the current change, you will skip the null dtos, however, not sure if any other parts of the code you were dependent on returning null if not found. You need to check if you are dependent on returning null or not at the Context.

File: src/main/java/io/hops/metadata/ndb/dalimpl/yarn/ResourceClusterJ.java [hunk 5]
```diff
@@ -145,10 +145,6 @@ public void add(Resource resourceNode) throws StorageException {
   }
 
   private Resource createHopResource(ResourceDTO resourceDTO) {
-    if (resourceDTO == null) {
```
Comment (maismail) [inline src/main/java/io/hops/metadata/ndb/dalimpl/yarn/ResourceClusterJ.java]: With the current change, you will skip the null dtos, however, not sure if any other parts of the code you were dependent on returning null if not found. You need to check if you are dependent on returning null or not at the Context.

File: src/main/java/io/hops/metadata/ndb/mysqlserver/CountHelper.java [hunk 6]
```diff
@@ -79,6 +81,19 @@ private static int count(String query) throws StorageException {
     } catch (SQLException ex) {
       throw new StorageException(ex);
     } finally {
+
+      if (s != null) {
+        try {
+          s.close();
+        } catch (SQLException ex) {
```
Comment (maismail) [inline src/main/java/io/hops/metadata/ndb/mysqlserver/CountHelper.java]: same exception issue

File: src/main/java/io/hops/metadata/ndb/mysqlserver/ScriptRunner.java [hunk 7]
```diff
@@ -124,93 +124,111 @@ public void runScript(Reader reader) throws IOException, SQLException {
...
+            command.append(line);
+            command.append("\n");
+          }
         } finally {
-            conn.rollback();
-            flush();
+          if (rs != null) {
+            try {
+              rs.close();
+            } catch (SQLException ex) {
```
Comment (maismail) [inline src/main/java/io/hops/metadata/ndb/mysqlserver/ScriptRunner.java]: same exception issue

File: src/main/java/io/hops/metadata/ndb/dalimpl/yarn/ResourceClusterJ.java (hunk 5 above)
Comment (berthoug) [reply src/main/java/io/hops/metadata/ndb/dalimpl/yarn/ResourceClusterJ.java]: createHopResource is called in findEntry and in createMap.
 - In findEntry it is called only if the dto is not null so this if will never be true.
 - In createMap it is followed by a hop.getId() which would throw a NullpointerException if we passed in this if.
So returning null is not supported by the current code.

**Pull request merged**